libkatana_python_native/include/katana/python/TemplateSupport.h

```diff
@@ -22,13 +28,41 @@ pybind11::object
 InstantiateForType(
     pybind11::module m, const std::string& basename, F f,
     pybind11::dict types) {
-  pybind11::object dtype = katana::PythonTypeTraits<T>::default_dtype();
-  pybind11::object cls = f.template instantiate<T>(
-      m, (basename + "[" + katana::PythonTypeTraits<T>::name + "]").c_str());
-  types[dtype] = cls;
+  pybind11::object representation =
+      katana::PythonTypeTraits<T>::representation();
+  pybind11::object cls =
+      f.template instantiate<T>(m, GetInstantiationName<T>(basename).c_str());
+  types[representation] = cls;
   return cls;
 }
 
+template <typename... Types, typename F>
+void
+InstantiateForTypes(pybind11::module m, const std::string& basename, F f) {
+  pybind11::module builtins = pybind11::module::import("builtins");
+  auto make_template_type1 = pybind11::cast<pybind11::function>(
+      pybind11::module::import("katana.native_interfacing.template_type")
+          .attr("make_template_type1"));
+  pybind11::dict types;
+
+  auto t = std::make_tuple(InstantiateForType<Types>(m, basename, f, types)...);
```

Comment:
You don't really need to make a tuple here - it's not used anywhere, as far I can see. Could you just fold over the comma operator instead?

Reply:
I don't think you can do that. Any expanded argument pack needs to be an argument to something else, I think. People often use arrays, I just used tuple here do that there is no need to have a dummy value of a consistent type (eg. `(code, 0)...`).

I'll give it a quick try though.

PS: Pedantry warning. But this is a map not a fold.